cincinnati/src/plugins/mod.rs

```diff
@@ -12,26 +12,25 @@ pub mod internal;
 pub use self::catalog::{build_plugins, deserialize_config, PluginSettings};
 use crate as cincinnati;
 use crate::plugins::interface::{PluginError, PluginExchange};
+use async_trait::async_trait;
 use failure::{Error, Fallible, ResultExt};
-use futures::IntoFuture;
-use futures::{Future, Stream};
 use std::collections::HashMap;
 use std::convert::{TryFrom, TryInto};
 use std::fmt::Debug;
 
 pub mod prelude {
-    pub use super::AsyncIO;
     pub use super::BoxedPlugin;
     pub use super::ExternalPluginWrapper;
     pub use super::InternalPluginWrapper;
+    pub use super::IO;
     pub use super::{build_plugins, deserialize_config, PluginSettings};
     pub use crate::{new_plugin, new_plugins};
     pub use futures_locks;
     pub use std::iter::FromIterator;
 }
 
-/// Convenience type to wrap other types in a Future
-pub type AsyncIO<T> = Box<dyn Future<Item = T, Error = Error> + Send>;
+/// Convenience type to wrap T
+pub type IO<T> = Fallible<T>;
```

Comment:
@lucab WDYT about this? I think this one could be either renamed to `Result<T>` or simply dropped and all occurrences replaced by `Fallible`

Reply:
:+1:

Reply:
It's gone!